soc/xtensa/intel_adsp/common/soc_mp.c

```diff
@@ -381,9 +344,6 @@ void arch_start_cpu(int cpu_num, k_thread_stack_t *stack, int sz,
 	k_busy_wait(10);
 	IDC[cpu_num].busy_int &= ~IDC_ALL_CORES;
 #endif
-
-	while (!start_rec.alive)
-		;
```

Comment:
wouldn't this let the primary core return early without waiting for the secondary core to boot and let the host start using the secondary core before it's ready?

Reply:
No, because that logic is already being done in the portable SMP layer in the Zephyr kernel.  arch_start_cpu() is not required to be atomic, this was just wasted bytes.